repose-aggregator/components/filters/ip-classification/src/test/scala/org/openrepose/filters/ipclassification/ConfigValidationTest.scala

```diff
@@ -0,0 +1,102 @@
+package org.openrepose.filters.ipclassification
+
+import org.junit.runner.RunWith
+import org.openrepose.filters.ipclassification.config.IpClassificationConfig
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.{FunSpec, Matchers}
+
+@RunWith(classOf[JUnitRunner])
+class ConfigValidationTest extends FunSpec with Matchers {
+
+  //Only testing that valid config is valid
+  // That the default works
+  // And that the XSD 1.1 assertions work
+
+  describe("verification of XSD validations") {
+    it("returns a config object when given a valid configuration") {
+      val validConfig =
+        """<?xml version="1.0" encoding="UTF-8"?>
+          |<ip-classification xmlns="http://docs.openrepose.org/repose/ip-classification/v1.0">
+          |    <user-header name="user-header" quality="0.5"/>
+          |    <group-header name="group-header" quality="0.5"/>
+          |    <group name="sample-group">
+          |        <cidr-ip>192.168.1.0/24</cidr-ip>
+          |        <cidr-ip>192.168.0.1/32</cidr-ip>
+          |    </group>
+          |    <group name="sample-ipv6-group">
+          |        <cidr-ip>2001:db8::/48</cidr-ip>
+          |    </group>
+          |    <group name="bolth-group">
+          |        <cidr-ip>10.10.220.0/24</cidr-ip>
+          |        <cidr-ip>2001:1938:80:bc::1/64</cidr-ip>
+          |    </group>
+          |    <group name="ipv4-match-all">
+          |        <cidr-ip>0.0.0.0/0</cidr-ip>
+          |    </group>
+          |</ip-classification>
+        """.stripMargin
+      val resultant = Marshaller.configFromString(validConfig)
+      resultant shouldNot be(null)
+      resultant.getClass should equal(classOf[IpClassificationConfig])
+
+      //Verify the group settings and such are not the defaults
+      resultant.getUserHeader.getName should equal("user-header")
+      resultant.getUserHeader.getQuality should equal(0.5D)
+      resultant.getGroupHeader.getName should equal("group-header")
+      resultant.getGroupHeader.getQuality should equal(0.5D)
+    }
+    //TODO: IF we want default true values, THEN we have to re-work where the defaults are specified; because XSD
```

Comment:
These are referring to the way XML handles (or doesn't) if an element is not present. That is what the other monkey business is about.